pkg/util/priorityupdate/priority_sort.go

```diff
@@ -0,0 +1,84 @@
+package priorityupdate
+
+import (
+	"sort"
+	"strconv"
+
+	appsv1alpha1 "github.com/openkruise/kruise/pkg/apis/apps/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// SortIndexesByPriority sort the indexes by UpdatePriorityStrategy
+func SortIndexesByPriority(strategy *appsv1alpha1.UpdatePriorityStrategy, pods []*v1.Pod, indexes []int) []int {
+	if strategy == nil || (len(strategy.WeightPriority) == 0 && len(strategy.OrderPriority) == 0) {
+		return indexes
+	}
+
+	f := func(i, j int) bool {
+		podI := pods[indexes[i]]
+		podJ := pods[indexes[j]]
+		wI, wJ := compare(strategy, podI.Labels, podJ.Labels)
+		if wI != wJ {
+			return wI >= wJ
+		}
+		return indexes[i] > indexes[j]
+	}
+
+	sort.Slice(indexes, f)
+	return indexes
+}
+
+func compare(strategy *appsv1alpha1.UpdatePriorityStrategy, podI, podJ map[string]string) (int64, int64) {
```

Comment:
Can compare function returns a bool?  This may improve readability.

Reply:
done